operator/pkg/helmreconciler/apply.go

```diff
@@ -211,4 +226,16 @@ func (h *HelmReconciler) ApplyObject(obj *unstructured.Unstructured) error {
 		}
 		return nil
 	})
+
+}
+
+// use server-side apply, require kubernetes 1.16+
+func (h *HelmReconciler) serverSideApply(obj *unstructured.Unstructured) error {
+	scope.Info("using server side apply to update obj")
+	objectStr := fmt.Sprintf("%s/%s/%s", obj.GetKind(), obj.GetNamespace(), obj.GetName())
+	opts := []client.PatchOption{client.ForceOwnership, client.FieldOwner(fieldOwnerOperator)}
```

Comment:
I think `kubectl` only does this with `--force-conflicts=false: If true, server-side apply will force the changes against conflicts.`. Is there any reason we should or should not set this?

Reply:
See comment at: https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/options.go#L655-L657. As istio operator manages the resources to be installed, so it makes sense for it to acquire ownership for conflicting fields, otherwise it cannot proceed during reconciliation if users have updated resources from other clients. And it point out that most controller should use that